### PR TITLE
add test infrastructure

### PR DESCRIPTION
**Pull request by fabiansinz**

### CODE REVIEW

**Comment by jgrewe** [comment]
nice!

**Comment by fabiansinz** [comment]
I can't get coveralls to work at the moment. Maybe someone else can try. At least the tests are working. The test-coverage is, of course, shamefully low.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+install:
+  - pip install git+https://github.com/G-Node/python-odml.git
+  - pip install -r requirements.txt
+  - pip install nose nose-cov python-coveralls
+
+# command to run tests
+script:
+  - nosetests -vv --with-coverage --cover-package=thunderfish
+after_success:
+  - coveralls
```
Comment (jgrewe) [inline .travis.yml:29]: If I am not mistaken we might need to add to the install commands
- coverage run --source=thunderfish setup.py test;

Comment (fabiansinz) [inline .travis.yml:29]: To install or after success?

Comment (jgrewe) [inline .travis.yml:29]: to install

**Comment by jgrewe** [comment]
grml ... 

travis says:

Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    packages=find_packages(exclude=['contrib', 'doc', 'tests*']),
NameError: name 'find_packages' is not defined

**Comment by fabiansinz** [comment]
Forgot to import the function. Next is on the way.

**Comment by fabiansinz** [comment]
Yes, coveralls works again!

**Comment by fabiansinz** [comment]
Yes! Coveralls works. Happy test writing. @jfsehuanes @tillraab if you want to try the tests locally, just copy the command from the `.travis.yml` (once the pull request is accepted).

```
nosetests -vv --with-coverage --cover-package=thunderfish
```

**Comment by jfsehuanes** [comment]
Thank you @fabiansinz  for your well known enthusiasm and efficiency. I however feel a little uncomfortable merging this pull request, since I really do not understand most of the stuff. I know what a test is meant to do (remember Fabian lecturing me on this) but I really do not know how exactly it works. Maybe @jgrewe should merge this pull-request?

**Comment by fabiansinz** [comment]
Just to be complete here. Basic testing is quite easy now:
- The directory `tests/` contains modules called `test_something.py` which contain functions called `test_somefeature` (the `test_` prefix is important).
- You can run the tests yourself by calling `nosetests -vv --with-coverage tests/` in the shell
- When you make a pull request [travis](travis-ci.org) runs the tests and reports back to github whether some of the failed
- After that [coveralls](http://coveralls.io/) checks how much of the code is covered by tests. You can look at what lines are covered by clicking on "Details" on the report by coveralls on github. 

I think the goal should be to reach at least 80% test coverage. Especially when restructuring the code it is very convenient because you can make sure that everything still runs as supposed after making changes.

**Pull request merged**